tests/fileVersionTest.js

```diff
@@ -77,46 +104,174 @@ describe('Main: Currently testing file versions management,', function () {
     expect(url).toBe(config.owncloudURL + 'remote.php/dav/meta/666/v/123456')
   })
 
-  it('retrieves file versions', function (done) {
-    oc.fileVersions.listVersions(versionedFileInfo.getFileId()).then(versions => {
-      expect(versions.length).toEqual(2)
-      expect(versions[0].getSize()).toEqual(2)
-      expect(versions[1].getSize()).toEqual(1)
-      oc.fileVersions.getFileVersionContents(versionedFileInfo.getFileId(), versions[0].getName()).then(content => {
-        expect(content).toBe('**')
-        oc.fileVersions.getFileVersionContents(versionedFileInfo.getFileId(), versions[1].getName()).then(content => {
-          expect(content).toBe('*')
-          done()
-        })
+  describe('file versions of non existing file', () => {
+    beforeAll(done => {
+      const promises = []
+      promises.push(setGeneralInteractions(provider))
+      promises.push(provider.addInteraction({
+        uponReceiving: 'PROPFIND file versions of non existent file',
+        withRequest: propfindFileVersionsRequestData,
+        willRespondWith: {
+          status: 404,
+          headers: applicationXmlResponseHeaders,
+          body: '<?xml version="1.0" encoding="utf-8"?>\n' +
+            '<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">\n' +
+            '  <s:exception>Sabre\\DAV\\Exception\\NotFound</s:exception>\n' +
+            '  <s:message/>\n' +
+            '</d:error>'
+        }
+      }))
+
+      Promise.all(promises).then(done, done.fail)
+    })
+
+    afterAll(function (done) {
+      provider.removeInteractions().then(done, done.fail)
+    })
+    fit('retrieves file versions of not existing file', function (done) {
+      oc.fileVersions.listVersions(fileInfo.id).then(versions => {
+        expect(versions).toBe(null)
+        done()
+      }).catch(error => {
+        expect(error.statusCode).toBe(404)
+        expect(error.message).toBe('')
+        done()
       })
     })
   })
 
-  it('retrieves file versions of not existing file', function (done) {
-    oc.fileVersions.listVersions(12345678).then(versions => {
-      expect(versions).toBe(null)
-      done()
-    }).catch(error => {
-      expect(error.statusCode).toBe(404)
-      expect(error.message).toBe('')
-      done()
+  describe('file versions for existing files', () => {
+    beforeAll(done => {
+      const promises = []
+      promises.push(setGeneralInteractions(provider))
+      promises.push(provider.addInteraction({
+        uponReceiving: 'PROPFIND file versions of existent file',
+        withRequest: propfindFileVersionsRequestData,
+        willRespondWith: {
+          status: 207,
+          headers: {
+            ...applicationXmlResponseHeaders,
+            'Access-Control-Allow-Headers': accessControlAllowHeaders,
+            'Access-Control-Allow-Methods': accessControlAllowMethods
```

Comment:
We can define this into a constant to make the code DRY as it is repeated in many other places as well.